stdlib/public/runtime/Errors.cpp

```diff
@@ -149,12 +149,15 @@ void swift::dumpStackTraceEntry(unsigned index, void *framePC,
     return;
   }
 
-  // If SymbolInfo:lookup succeeded then fileName is non-null. Thus, we find the
+  // If SymbolInfo:lookup succeeded and fileName is non-null, we can find the
   // library name here. Avoid using StringRef::rsplit because its definition
   // is not provided in the header so that it requires linking with
   // libSupport.a.
-  llvm::StringRef libraryName{syminfo->getFilename()};
-  libraryName = libraryName.substr(libraryName.rfind('/')).substr(1);
+  llvm::StringRef libraryName{"<unavailable>"};
+  if (const char *filename = syminfo->getFilename()) {
+    libraryName = filename;
+    libraryName = libraryName.substr(libraryName.rfind('/')).substr(1);
```

Comment:
Hmm, is there a guarantee that the path is in Unix spelling?

Reply:
Not even remotely, no!

I could split `getFilename()` into `getLibraryPath()` which returns the full string, and `getFilename()` which returns the last path component, and then use whatever Windows-ism is appropriate in `getFilename()`. Then the logic doesn't need to be out here in Errors.cpp. Does that sound good?